### PR TITLE
Glimpse2 splitref

### PR DESCRIPTION
**Pull request by LouisLeNezet**
## PR checklist

Here is the next modules of GLIMPSE2 software.
This modules allow to split a reference panel in a bin format.
Preferably to use after GLIMPSE2_chunk.

- [X] This comment contains a description of changes (with reason).
- [X] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [X] If necessary, include test data in your PR.
- [X] Remove all TODO statements.
- [X] Emit the `versions.yml` file.
- [X] Follow the naming conventions.
- [X] Follow the parameters requirements.
- [X] Follow the input/output options guidelines.
- [X] Add a resource `label`
- [X] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [X] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/glimpse2/splitreference/main.nf [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    """
+
+    conda "bioconda::glimpse-bio=2.0.0"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/glimpse-bio:2.0.0--hf340a29_0':
+        'quay.io/biocontainers/glimpse-bio:2.0.0--hf340a29_0' }"
+
+    input:
+        tuple val(meta) , path(reference), path(reference_index), val(input_region), val(output_region)
+        path(map)
```
Comment (nvnieuwk) [inline modules/nf-core/glimpse2/splitreference/main.nf:22]:
```suggestion
        tuple val(meta2), path(map)
```

File: modules/nf-core/glimpse2/splitreference/main.nf [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+
+    output:
+        tuple val(meta), path("*.bin"), emit: bin_ref
+        path "versions.yml"       , emit: versions
```
Comment (nvnieuwk) [inline modules/nf-core/glimpse2/splitreference/main.nf:27]:
```suggestion
        tuple val(meta), path("*.bin"), emit: bin_ref
        path "versions.yml"           , emit: versions
```
Improves readability :)

File: tests/modules/nf-core/glimpse2/splitreference/main.nf [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+nextflow.enable.dsl = 2
+
+include { GLIMPSE2_SPLITREFERENCE } from '../../../../../modules/nf-core/glimpse2/splitreference/main.nf'
+
+workflow test_glimpse2_splitreference {
+    input = [
+        [ id:'ref1000GP', single_end:false ], // meta map
+        file("https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/glimpse/1000GP.chr21.noNA12878.s.bcf", checkIfExists: true),
+        file("https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/glimpse/1000GP.chr21.noNA12878.s.bcf.csi", checkIfExists: true),
```
Comment (nvnieuwk) [inline tests/modules/nf-core/glimpse2/splitreference/main.nf:11]: Doesn't the module work with the VCF files already in the test-datasets repo?

File: tests/modules/nf-core/glimpse2/splitreference/main.nf [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+        "chr21:16600000-16800000",
+        "chr21:16600000-16800000"]
+    GLIMPSE2_SPLITREFERENCE (input, [])
```
Comment (nvnieuwk) [inline tests/modules/nf-core/glimpse2/splitreference/main.nf:14]:
```suggestion
    GLIMPSE2_SPLITREFERENCE (input, [[],[]])
```
Can you also add a test with the map?

**Review by nvnieuwk — CHANGES_REQUESTED** [review]
Overall very good, here are some small suggestions :)

File: tests/modules/nf-core/glimpse2/splitreference/main.nf (hunk 3 above)
Comment (nvnieuwk) [reply tests/modules/nf-core/glimpse2/splitreference/main.nf:11]: Allright :)

**Review by nvnieuwk — APPROVED** [review]

**Comment by LouisLeNezet** [comment]
Thanks again @nvnieuwk !

**Pull request merged**